src/stan/mcmc/hmc/static_uniform/adapt_diag_e_static_uniform.hpp

```diff
@@ -31,10 +31,15 @@ class adapt_diag_e_static_uniform
       this->stepsize_adaptation_.learn_stepsize(this->nom_epsilon_,
                                                 s.accept_stat());
 
-      bool update = this->var_adaptation_.learn_variance(this->z_.inv_e_metric_,
-                                                         this->z_.q);
+      Eigen::VectorXd inv_metric;
```

Comment:
Can you know the size of this when you make it?

Reply:
I could, but it gets resized in `learn_variance`. Should I still set the size when declaring it?

Reply:
If you declare the sizes there then the resize is a no-op since it's already the correct size. But idt it's that big of a deal either way

Reply:
I just left it off. It's kindof a weird thing happening here anyway (like `learn_variance` is effectively returning two things).